Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1

```diff
@@ -0,0 +1,761 @@
+
+                # Assert
+                $dbConnectionString.DbName | Should -Be "EDWAdmin2"
+                $dbConnectionString.DbConnectionString | Should -Be "Server=host.fabric.local;Database=EDWAdmin2;Trusted_Connection=True;MultipleActiveResultSets=True;"
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Read-Host -Times 1 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Invoke-Sql -Times 1 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Add-InstallationSetting -Times 1 -Exactly
+            }
+        }
+    }
+}
+
+Describe 'Publish-Identity'{
+    Context 'App pool exists'{
+        InModuleScope Install-Identity-Utilities{
+            It 'Should install app and return version and directory'{
+                # Arrange
+                $userName = "fabric\test.user"
+                $password = ConvertTo-SecureString "supersecretpassword" -AsPlainText -Force
+                $credential = New-Object -TypeName "System.Management.Automation.PSCredential" -ArgumentList $userName, $password
+                $expectedVersion = "1.4.12345"
+
+                $site = @{Name="Default Web Site"; physicalPath = "C:\inetpub\wwwroot"}
+                $iisUser = @{UserName = $userName; Credential = $credential }
+
+                
+                Mock -ModuleName Install-Identity-Utilities -CommandName New-AppRoot -MockWith {}
+                Mock -ModuleName Install-Identity-Utilities -CommandName Test-AppPoolExistsAndRunsAsUser -MockWith { $true }
+                Mock -ModuleName Install-Identity-Utilities -CommandName New-App -MockWith {}
+                Mock -ModuleName Install-Identity-Utilities -CommandName Publish-WebSite -MockWith {}
+                Mock -ModuleName Install-Identity-Utilities -CommandName New-AppPool -MockWith {}
+                Mock -ModuleName Install-Identity-Utilities -CommandName Get-InstalledVersion -MockWith { $expectedVersion }
+
+                # Act
+                $publishedApp = Publish-Identity -site $site -appName "identity" -iisUser $iisUser -zipPackage "$env:Temp\identity.zip"
+
+                # Assert
+                $publishedApp.version = "1.4.12345"
+                $publishedApp.applicationDirectory = "C:\inetpub\wwwroot\identity"
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName New-AppRoot -Times 1 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName New-App -Times 1 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Publish-WebSite -Times 1 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName New-AppPool -Times 0 -Exactly
+
+            }
+        }
+    }
+
+    Context 'App pool does not exists'{
```

Comment:
should this be App pool exists?

Reply:
I think this is correct, if the app pool doesn't already exists, we are verifying that we do call the new app pool function. The above context tests the reverse, we validate that if the app pool does exist that we do call the new-apppool function.